graph/minimum_spanning_tree_with_disjoint_set_union.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file
+ * @brief [Disjoint union](https://cp-algorithms.com/data_structures/disjoint_set_union.html)
+ *
+ * @details
+ * This data structure provides the following capabilities. We are given several elements, each of which is a separate set. A DSU will have an operation to combine any two sets, and it will be able to tell in which set a specific element is. The classical version also introduces a third operation, it can create a set from a new element.
+ *
+ * ### Algorithm
+ * In Graph, if you have to find out the minimum spanning tree, i.e. a subgraph which is actually a tree and total weight of that tree is minimum, then DSU could be used.
+ *
+ * @author [Aman Agrawal](https://github.com/amanagr6697)
+ */
+/** Optimizations:
+* 1. Works like assigning main parent to all intermediate nodes while unwinding from recursion
+* 2. Here merge happens by height that is if one connected component height's is x, another is y and x>y then while merging x and y, y would get attached to x
+* These 2 optimzations make DSU way faster making it runable in approximately amorized constant time
+**/
+#include <iostream> //for io
+#include<vector>    //for vector
+#include <algorithm> //for sorting
+using namespace std;
```

Comment:
This is discouraged. Please strictly use `std::` for each member. You can also use `using std::cout;` and so on for each `std::` member you want to shorten. 🙂